src/components/MoviesList/MoviesList.tsx

```diff
@@ -4,12 +4,28 @@ import { MovieCard } from '../MovieCard';
 
 interface Props {
   movies: Movie[];
+  selectedMovies: string;
 }
 
-export const MoviesList: React.FC<Props> = ({ movies }) => (
-  <div className="movies">
-    {movies.map(movie => (
-      <MovieCard key={movie.imdbId} movie={movie} />
-    ))}
-  </div>
-);
+export const MoviesList: React.FC<Props> = ({ movies, selectedMovies }) => {
+  const filtered = (allMovies: Movie[]) => {
+    return allMovies.filter(e => {
+      const lowerCaseOrigin = e.title.toLocaleLowerCase();
+      const lowerCaseSelected = selectedMovies.toLocaleLowerCase();
+
+      return lowerCaseOrigin.includes(lowerCaseSelected);
+    });
+  };
+
+  return (
+    <div className="movies">
+      {selectedMovies
+        ? filtered(movies).map(movie => (
+          <MovieCard key={movie.imdbId} movie={movie} />
+        ))
+        : movies.map(movie => (
+          <MovieCard key={movie.imdbId} movie={movie} />
+        ))}
```

Comment:
Refactor. Create visibleMovies variable and use it for render. If query is not empty it will be equal movies or filtered movies if query is not empty